plugins/crypto/src/main/java/org/opensearch/cryptoplugin/cache/CacheRefresher.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.cryptoplugin.cache;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.AbstractAsyncTask;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class CacheRefresher extends AbstractAsyncTask {
+    private static final Logger LOG = LogManager.getLogger(CacheRefresher.class);
+    private final String threadpoolName;
+
+    private final Set<CacheRefreshable> refreshables;
+
+    public CacheRefresher(String threadpoolName, ThreadPool threadpool) {
+        super(LOG, threadpool, TimeValue.timeValueMinutes(1), true);
+        this.threadpoolName = threadpoolName;
+        refreshables = Collections.synchronizedSet(new HashSet<>());
+        rescheduleIfNecessary();
+    }
+
+    public synchronized void register(CacheRefreshable refreshable) {
```

Comment:
How do you envision these methods being used? We want to be cautious with synchronization to avoid bottlenecks and it seems like this would be a bootstrapping process? 

Nit: Why not stick with standard set terminology i.e. add & remove?

Reply:
No, this is not during bootstrap process. It is invoked during repo registration. Since repo registration is not a frequent event, I went ahead with synchronized constructs to avoid race in registering an instance on cache.